tests/test_examples.py

```diff
@@ -0,0 +1,78 @@
+import itertools
+import sys
+from pathlib import Path
+from subprocess import PIPE  # noqa: S404
+from subprocess import Popen  # noqa: S404
+from subprocess import STDOUT  # noqa: S404
+from time import sleep
+from typing import List
+
+import pytest
+
+HERE = Path(sys.argv[0] if __name__ == "__main__" else __file__).resolve().parent
+
+
+# FIXTURES
+
+
+@pytest.fixture
+def path_1_simple() -> Path:
+    path = (HERE / ".." / "examples" / "1_simple").resolve()
+    assert path.exists()
+    return path
+
+
+@pytest.fixture
+def path_2_base_time_add() -> Path:
+    path = (HERE / ".." / "examples" / "2_base_time_add").resolve()
+    assert path.exists()
+    return path
+
+
+# UTILS
+
+
+def assert_run_in_parallel(scrips_to_run: List[Path]) -> None:
+    # check all exist before running
+    for script_to_run in scrips_to_run:
+        assert script_to_run.exists()
+
+    for permutation in itertools.permutations(scrips_to_run):
+        # run provided scripts
+        print(f"Running permutation {permutation}")
+        processes = [
+            Popen(  # noqa: S607
+                ["python", f"{x}"], shell=True, stdout=PIPE, stderr=STDOUT  # noqa: S602
+            )
+            for x in permutation
+        ]
+
+        # wait for processes to finish
+        continue_running = True
+        while continue_running:
+            continue_running = all(
+                [process.returncode is not None for process in processes]
+            )
+            sleep(0.3)
+
+        # ensure all processes finished successfully
+        for process in processes:
+            stdout, _ = process.communicate()
```

Comment:
but these do not run in parallel as you claim in ``assert_run_in_parallel`` , right?

Reply:
Yes they do run in parallel, otherwise it would not be possible for `example/2_base_time_add` to finish.
In the above loop I wait for the process to provide a `returncode`, meaning it finished execution.

Here I fetch the standard output.